### PR TITLE
fix verbose printing of proxies in 6.26

### PR DESCRIPTION
**Pull request by will-cern**
# This Pull request:

Discovered that in 6.26 verbose print of lots of classes doesn't work. e.g. you can do:

```
RooWorkspace ws;
ws.factory("SUM::model(RooExponential::pdf1(obs1[0,5],tau1[-1]))"); 
ws.pdf("model")->Print("v")
```

and it will crash.

## Changes or fixes:
Fixes verbose printout of roofit object that uses RooListProxy (lots of classes do). 

## Checklist:

- [x ] tested changes locally
- [x ] updated the docs (if necessary)

This PR fixes #

### CODE REVIEW

**Review by hageboeck — CHANGES_REQUESTED** [review]
Hey Will,
good idea! I added a suggestion to save a few CPU cycles

File: roofit/roofitcore/src/RooAbsArg.cxx [hunk 1]
```diff
@@ -1541,12 +1541,12 @@ void RooAbsArg::printMultiline(ostream& os, Int_t /*contents*/, Bool_t /*verbose
...
 	os << " (empty)" << endl ; ;
       }
-    } else {
+    } else if(dynamic_cast<RooAbsCollection*>(proxy)) {
       os << indent << "    " << proxy->name() << " -> " ;
       os << endl ;
       TString moreIndent(indent) ;
       moreIndent.Append("    ") ;
-      ((RooSetProxy*)proxy)->printStream(os,kName,kStandard,moreIndent.Data()) ;
+      dynamic_cast<RooAbsCollection*>(proxy)->printStream(os,kName,kStandard,moreIndent.Data()) ;
```
[the remark this answers is not in the document]
Comment (will-cern) [reply roofit/roofitcore/src/RooAbsArg.cxx]: thanks - I was too lazy to check if there was any virtual inheritance anywhere but I trust you that there isn't!

[the remark this answers is not in the document]
Comment (hageboeck) [reply roofit/roofitcore/src/RooAbsArg.cxx]: Oh, there might be, but the compiler will complain if you cannot cross-cast. 🙂

**Comment by hageboeck** [comment]
@phsft-bot build

File: roofit/roofitcore/src/RooAbsArg.cxx (hunk 1 above)
[the remark this answers is not in the document]
Comment (hageboeck) [reply roofit/roofitcore/src/RooAbsArg.cxx:1549]: OK, I'll take it back! 🙂

**Comment by hageboeck** [comment]
@will-cern Given that my idea didn't work, can we clean up the commit history using an interactive rebase or `git reset --hard <commit-hash>` to make the two extra commits disappear? It's safe to force-push to your current branch, so we merge only the first commit.

**Comment by will-cern** [comment]
do MR not have their commits squashed?

**Comment by hageboeck** [comment]
> do MR not have their commits squashed?

That's a viable option, indeed. 🙂 
@guitargeek Ready to merge (+squash)?

**Review by hageboeck — APPROVED** [review]
LGTM, but please squash on merge.

**Comment by hageboeck** [comment]
@will-cern thanks for the fix!

**Pull request merged**